core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -15,8 +15,12 @@
 import alluxio.Configuration;
 import alluxio.Constants;
 import alluxio.PropertyKey;
+import alluxio.exception.ExceptionMessage;
```

Comment:
I guess I would like for the UFS to be simple to create new connections without having to worry too much about correctness. If UFS would default to the correct, but inefficient method of temp file rename, then creating a new UFS would be easier to reason about. UFSes with atomic creates can take advantage of the optimization without renaming.

However, I don't know how best to organize the classes to do this effectively. In the original functionality, the UFSes could just use the `FileOutStream` to do the renaming for it. Is there a simple way to let the out stream do the renaming, but also have an outstream that does not do the renaming (direct create)?

What do you guys think?